components/molecule/autosuggest/src/index.js

```diff
@@ -26,67 +26,57 @@ const getIsTypeableKey = key => {
   return key.length === 1 || keysEdit.includes(key)
 }
 
-class MoleculeAutosuggest extends Component {
-  refMoleculeAutosuggest =
-    this.props.refMoleculeAutosuggest || React.createRef()
-
-  refsMoleculeAutosuggestOptions = []
-
-  refMoleculeAutosuggestInput = React.createRef()
-
-  state = {
-    focus: false
-  }
-
-  get extendedChildren() {
-    const {children, keysSelection} = this.props // eslint-disable-line react/prop-types
-    const {refsMoleculeAutosuggestOptions} = this
-    return React.Children.toArray(children)
-      .filter(Boolean)
-      .map((child, index) => {
-        refsMoleculeAutosuggestOptions[index] = React.createRef()
-        return React.cloneElement(child, {
-          innerRef: refsMoleculeAutosuggestOptions[index],
-          onSelectKey: keysSelection
-        })
+const MoleculeAutosuggest = ({multiselection, ...props}) => {
+  const {
+    refMoleculeAutosuggest: refMoleculeAutosuggestFromProps,
+    children,
+    onToggle,
+    onChange,
+    isOpen,
+    keysCloseList,
+    keysSelection
+  } = props
+
+  const refMoleculeAutosuggest = useRef(refMoleculeAutosuggestFromProps)
```

Comment:
Doesn't `refMoleculeAutosuggestFromProps` have its own `current` property when present? I'd suggest:
```js
const refMoleculeAutosuggest = refMoleculeAutosuggestFromProps || useRef()
```

I think that's the most similar functional approach to substitute the original class approach, isn't it?
```js
refMoleculeAutosuggest =
    this.props.refMoleculeAutosuggest || React.createRef()
```